### PR TITLE
chore(set-react-version): apply patches while setting version

### PR DESCRIPTION
**Pull request by tido64**
### Description

Apply patches while setting version.

### Platforms affected

- [ ] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows

### Test plan

1. Verify that `gradle-wrapper.properties` gets patched: `npm run set-react-version 0.72`
2. Verify that Jetifier gets disabled (`example/android/gradle.properties`): `npm run set-react-version nightly`

### CODE REVIEW

File: scripts/set-react-version.mjs [hunk 1]
```diff
@@ -97,6 +97,20 @@ async function checkEnvironment() {
   return true;
 }
 
+/**
+ * Disables [Jetifier](https://developer.android.com/tools/jetifier).
```
Comment (kelset) [inline scripts/set-react-version.mjs:101]: might be worth adding an extra line to reference why from 74 onward we don't want to use jetifier anymore

File: scripts/set-react-version.mjs [hunk 2]
```diff
@@ -417,18 +448,24 @@ export function setReactVersion(version, coreOnly) {
...
+  } else {
+    setReactVersion(version, process.argv.includes("--core-only")).then(() => {
+      const numVersion =
+        version === "nightly"
+          ? Number.MAX_SAFE_INTEGER
+          : toVersionNumber(version);
+      if (numVersion >= v(0, 74, 0)) {
+        disableJetifier();
+      } else if (numVersion < v(0, 73, 0)) {
+        setGradleVersion("7.6.3");
```
Comment (kelset) [inline scripts/set-react-version.mjs:471]: based on my testing for the other PR -> https://github.com/microsoft/react-native-test-app/pull/1837#issuecomment-1941437582

we still have the webstorage webapi problem when downgrading and testing Android; do we want/need to have a patch similar to the one we had for safearea to apply here? we can leave it for a follow up PR if yes

**Review by kelset — APPROVED** [review]
tested locally, all works. Add a couple of comments but they are low prior/not blocking 👍

File: scripts/set-react-version.mjs (hunk 2 above)
Comment (tido64) [reply scripts/set-react-version.mjs:471]: We should fix this in a separate PR. I'm prepping some fixes in `web-storage`.

**Pull request merged**